grakn-graph/src/main/java/ai/grakn/graph/internal/AbstractGraknGraph.java

```diff
@@ -676,78 +677,20 @@ private CastingImpl getCasting(RoleTypeImpl role, InstanceImpl rolePlayer){
         }
     }
 
-    private void putShortcutEdges(RelationImpl relation, RelationType relationType, CastingImpl newCasting){
-        Map<RoleType, Set<Instance>> roleMap = relation.allRolePlayers();
-
-        //This ensures the latest casting is taken into account when transferring relations
-        roleMap.computeIfAbsent(newCasting.getRole(), (k) -> new HashSet<>()).add(newCasting.getRolePlayer());
-
-        Set<CastingImpl> castings = relation.getMappingCasting();
+    private void putNewShortcutEdge(Instance fromInstance, Relation toRelation, RoleType roleType){
+        boolean exists  = getTinkerPopGraph().traversal().V(fromInstance.getId().getRawValue()).
```

Comment:
It may be wiser to start from `toRelation`, simply because it _probably_ has fewer edges.